### PR TITLE
Implement Prometheus Service Endpoint

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

**Comment by deadeyegoodwin** [comment]
@GuanLuo You need to rebase to master and change to use the cmake build.

**Comment by GuanLuo** [comment]
rebased

File: src/servers/http_server.cc [hunk 1]
```diff
@@ -574,9 +627,25 @@ HTTPServer::Create(
...
     LOG_INFO << "Starting HTTPService at " << addr;
     http_servers->emplace_back(
-        new HTTPServerImpl(server, ep_map.second, ep_map.first, thread_cnt));
+        new HTTPAPIServer(server, ep_map.second, ep_map.first, thread_cnt));
   }
 
   return Status::Success;
 }
+
+Status
```
Comment (deadeyegoodwin) [inline src/servers/http_server.cc]: Can we change Create() to CreateAPIServer() to keep the function names consistent.

File: src/core/metrics.cc [hunk 2]
```diff
@@ -367,6 +368,14 @@ Metrics::GetRegistry()
   return singleton->registry_;
 }
 
+const std::string
+Metrics::SerializedMetrics()
+{
+  auto singleton = Metrics::GetSingleton();
+  return singleton->serializer_->Serialize(
```
Comment (deadeyegoodwin) [inline src/core/metrics.cc]: Is there a serializer that doesn't include all the comments in the outout?

Comment (GuanLuo) [reply src/core/metrics.cc]: No, this is the only serializer provided by prometheus-cpp and there is no option to exclude the comments. To do so, we can either strip the comments out from the serialized metrics or provide our own serializer implementation.

**Pull request merged**